File: dkb/convert_dkb_date_test.go
package dkb

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestConvertDkbDate_HappyPath(t *testing.T) {
  input := "02.01.23"

  output, _ := convertDkbDate(input)
  expectedDate := "2023-01-02"

	assert.Equal(t, expectedDate, output)
}

func TestConvertDkbDate_SadPath(t *testing.T) {
  input := "some.invalid.date"

  _, err := convertDkbDate(input)
  expectedRegex := "^.*cannot parse.*$"

  assert.Regexp(t, expectedRegex, err.Error())
}

File: dkb/convertDkbDate.test.ts
import { describe, it } from "@std/testing/bdd";
import { expect } from "@std/expect";
import { convertDkbDate } from "./convertDkbDate.ts";

describe("convertDkbDate", () => {
  it("converts date from dd.mm.yy to yyyy-mm-dd", () => {
    const input = "02.01.23";
    const expectedDate = "2023-01-02";
    expect(convertDkbDate(input)).toEqual(expectedDate);
  });

  it("throws error for invalid date format", () => {
    const input = "some.invalid.date";
    expect(() => convertDkbDate(input)).toThrow();
  });
});
